var/spack/repos/builtin/packages/gnutls/package.py

```diff
@@ -64,6 +65,7 @@ def configure_args(self):
 
         if '+guile' in spec:
             args.append('--enable-guile')
+            os.environ['GUILE'] = join_path(spec["guile"].prefix.bin, 'guile')
```

Comment:
I think it's worth moving this into `setup_environment`. I mean something like:
```python
def setup_environment(self, build_env, run_env):
    if '+guile' in self.spec:
        build_env.set('GUILE', os.path.join(spec["guile"].prefix.bin, 'guile'))
```

Reply:
@alalazo that seems like the proper thing to do. Commit pushed. Please let me know if this looks good now.